### PR TITLE
Integrating DeltaLake

### PR DESCRIPTION
**Pull request by aamend**
Signed-off-by: antoine.amend <antoine.amend@databricks.com>

### CODE REVIEW

**Comment by aamend** [comment]
@maintainer, please let me know if additional info is needed to trigger a running workflow

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/protocols/pure/v1_20_0/models/metamodel_connection.pure [hunk 1]
```diff
@@ -33,6 +33,11 @@ Class meta::protocols::pure::v1_20_0::metamodel::store::relational::connection::
 {
 }
 
+Class meta::protocols::pure::v1_20_0::metamodel::store::relational::connection::alloy::authentication::DeltaLakeAuthenticationStrategy extends meta::protocols::pure::v1_20_0::metamodel::store::relational::connection::alloy::authentication::AuthenticationStrategy
```
Comment (epsstan) [inline legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/protocols/pure/v1_20_0/models/metamodel_connection.pure:36]: See comments in the legend-engine PR for modeling this as a vault reference etc

File: README.md [hunk 2]
```diff
@@ -27,3 +27,4 @@ Copyright 2020 Goldman Sachs
 Distributed under the [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0).
 
 SPDX-License-Identifier: [Apache-2.0](https://spdx.org/licenses/Apache-2.0)
+
```
Comment (epsstan) [inline README.md:30]: Are there any considerations/gotchas around 
1/ Case sensitivity of database/table/column identifiers 
2/ Quoting of identifiers to support names with spaces, special chars 
3/ Special handling for database schemas etc. For e.g in Spanner, you cannot have a non default schema. So I'm thinking of throwing an error if a model has a non default schema 
4/ Any SQL compatibility issues to be tested. See testBigQueryCompatibility,pure and other pure tests under legend-pure-code-compiled-core-relational

Comment (epsstan) [reply README.md:30]: This PR introduces the new database type into the latest V_X__X_X version. Once this PR is merged, the changes need to also be introduced in the versioned protocols. See v_120_0 etc

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/SQLQueryToString.pure [hunk 3]
```diff
@@ -1059,7 +1059,9 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::getDy
                                                  choice(DatabaseType.Postgres,     $allStates,            ^ToSql(format='(%s + (INTERVAL \'1 %s\' * %s))', transform={p:String[3] | [$p->at(0), $p->at(2)->replace('Text', ''), $p->at(1)]})), //https://www.postgresql.org/message-id/2B704E31-B533-4159-B6F0-EC284CFE824C%40skype.net
```
Comment (epsstan) [inline legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/SQLQueryToString.pure:1059]: Can we add tests for the functions added here ?

File: legend-pure-ide-light/src/main/web/yarn.lock [hunk 4]
```diff
@@ -2,36 +2,29 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@7.12.11":
-  version "7.12.11"
-  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.12.11.tgz#f4ad435aa263db935b8f10f2c552d23fb716a63f"
-  integrity sha512-Zt1yodBx1UcyiePMSkWnU4hPqhwq7hGi2nFL1LeA3EUl+q2LQx16MISgJ0+z7dnmgvP9QtIleuETGOiOH1RcIw==
+"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.12.13", "@babel/code-frame@^7.8.3":
```
Comment (epsstan) [inline legend-pure-ide-light/src/main/web/yarn.lock:5]: Is this all whitespace and formatting changes ? If yes, can it be removed from the PR ?

**Comment by epsstan** [comment]
Is it possible to run a local integration test for the DeltaLake connector. For e.g can we run a test with something that implements the AWS S3 protocol ? Also, not sure how to run the test if we can't resolve or have the driver in the repo.

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/protocols/pure/v1_20_0/models/metamodel_connection.pure (hunk 1 above)
Comment (aamend) [reply legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/protocols/pure/v1_20_0/models/metamodel_connection.pure]: Created an authentication strategy generic enough for `APIToken` - see latest commit

File: README.md (hunk 2 above)
Comment (aamend) [reply README.md]: - I handled back quotes of table / column names and could not spot any special complexity to handle
- unit tested all functions
- ported changes to `v_120_0` as well

File: legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/SQLQueryToString.pure (hunk 3 above)
Comment (aamend) [reply legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/SQLQueryToString.pure:1094]: Added

File: legend-pure-ide-light/src/main/web/yarn.lock (hunk 4 above)
Comment (aamend) [reply legend-pure-ide-light/src/main/web/yarn.lock]: This is actually code generated at compile time. We should consider adding to `.gitignore`

**Pull request merged**